thrift/types.go

```diff
@@ -100,15 +106,67 @@ func parseDouble(value interface{}) (float64, error) {
 	return f64, nil
 }
 
+// parseBinaryList will try to parse a list of numbers
+// or a list of strings.
+func parseBinaryList(vl []interface{}) ([]byte, error) {
+	bs := make([]byte, 0, len(vl))
+	for _, v := range vl {
+		switch v := v.(type) {
+		case json.Number:
+			vInt, err := strconv.ParseInt(v.String(), 10, 8)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse list of bytes: %v", err)
+			}
+			bs = append(bs, byte(vInt))
+		case string:
+			if len(v) != 1 {
+				return nil, fmt.Errorf("not a valid character in list of characters: %q", v)
+			}
+			bs = append(bs, v[0])
```

Comment:
If you remove this restriction and append `append(bs, v...)`, the following should just work, right?

```
{"somefield": [1, "a", 5, "foo"]}
```

Any reason to not support that?